rpc/core/blocks.go

```diff
@@ -81,6 +81,16 @@ func Block(ctx *rpctypes.Context, heightPtr *int64) (*ctypes.ResultBlock, error)
 	return &ctypes.ResultBlock{BlockID: blockMeta.BlockID, Block: block}, nil
 }
 
+// BlockByHash gets block of given hash.
+// More: https://tendermint.com/rpc/#/Info/block_by_hash
+func BlockByHash(ctx *rpctypes.Context, hash []byte) (*ctypes.ResultBlock, error) {
+	block := blockStore.LoadBlockByHash(hash)
+	height := block.Height
+
+	blockMeta := blockStore.LoadBlockMeta(height)
```

Comment:
easier to construct BlockMeta from block than to load it.

Reply:
@melekes Everywhere in Block and Commit, we are using LoadBlockMeta to get the BlockMeta. So, Should I change this? I thought it will make consistent code.
Also, To construct blockMeta we also need BlockParts along with blocks, I guess. Correct me if I am wrong here.

Reply:
Right, thanks!